File: market/test/java/ru/yandex/market/ff4shops/repository/PartnerRepositoryTest.java
package ru.yandex.market.ff4shops.repository;

import java.util.List;
import java.util.Optional;
import java.util.Set;
import java.util.stream.Collectors;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.support.TransactionTemplate;

import ru.yandex.market.common.test.db.DbUnitDataSet;
import ru.yandex.market.core.feature.model.FeatureType;
import ru.yandex.market.ff4shops.config.FunctionalTest;
import ru.yandex.market.ff4shops.mbi.feature.model.FeatureStatus;
import ru.yandex.market.ff4shops.partner.dao.PartnerRepository;
import ru.yandex.market.ff4shops.partner.dao.model.PartnerEntity;

import static java.util.Arrays.asList;
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class PartnerRepositoryTest extends FunctionalTest {

    @Autowired
    private PartnerRepository tested;
    @Autowired
    private TransactionTemplate transactionTemplate;

    @Test
    @DbUnitDataSet(after = "SupplierRepositoryTest.insertSuppliers.after.csv")
    void testInsert() {
        PartnerEntity entity1 = new PartnerEntity(1, 100L, true, FeatureType.MARKETPLACE, FeatureStatus.SUCCESS, true,
                true, null, true);
        PartnerEntity entity2 = new PartnerEntity(2, 100L, true, FeatureType.MARKETPLACE, FeatureStatus.NEW, true,
                false, null, false);

        transactionTemplate.execute(status -> {
            tested.insert(asList(entity1, entity2));
            return null;
        });
    }

    /**
     * Проверяем, что берутся только записи с запрошенными id и с push_stocks = true
     */
    @Test
    @DbUnitDataSet(before = "PartnerRepositoryTest.testFindAllWithPushStocks.before.csv")
    void testFindAllWithPushStocks() {
        List<Long> actualPartners = tested.findAllWithPushStocks(Set.of(1L, 2L, 3L)).stream()
                .map(PartnerEntity::getId)
                .collect(Collectors.toList());

        assertEquals(2, actualPartners.size());
        assertTrue(actualPartners.containsAll(Set.of(1L, 3L)));
    }

    /**
     * Тест простого поиска по ID
     */
    @Test
    @DbUnitDataSet(before = "PartnerRepositoryTest.testFindById.before.csv")
    void testFindById() {
        Optional<PartnerEntity> partner = tested.findById(1L);


        PartnerEntity expected = new PartnerEntity();
        expected.setId(1);
        expected.setBusinessId(1L);
        expected.setEnabled(true);
        expected.setFeatureStatus(FeatureStatus.SUCCESS);
        expected.setFeatureType(FeatureType.DROPSHIP);
        expected.setCpaPartnerInterface(true);
        expected.setPushStocks(true);
        expected.setStocksByPartnerInterface(false);

        assertThat(partner.isPresent());
        assertThat(partner.get()).isEqualTo(expected);
    }

    /**
     * Тест простого поиска по ID
     */
    @Test
    @DbUnitDataSet(before = "PartnerRepositoryTest.testFindByIdWithNullStockByPiFlag.before.csv")
    void testFindByIdWithNullStockByPiFlag() {
        Optional<PartnerEntity> partner = tested.findById(1L);


        PartnerEntity expected = new PartnerEntity();
        expected.setId(1);
        expected.setBusinessId(1L);
        expected.setEnabled(true);
        expected.setFeatureStatus(FeatureStatus.SUCCESS);
        expected.setFeatureType(FeatureType.DROPSHIP);
        expected.setCpaPartnerInterface(true);
        expected.setPushStocks(true);

        assertThat(partner.isPresent());
        assertThat(partner.get()).isEqualTo(expected);
    }

    @Test
    @DbUnitDataSet(before = "PartnerRepositoryTest.testUpdateStocksByPiFlag.before.csv",
            after = "PartnerRepositoryTest.testUpdateStocksByPiFlag.after.csv")
    void testUpdateStocksByPiFlag() {
        tested.updateStocksByPiFlag(1, true);
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/orderservice/tms/persistence/repository/PartnerRepositoryTest.kt
package ru.yandex.market.mbi.orderservice.tms.persistence.repository

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import ru.yandex.market.mbi.orderservice.common.enum.PlacementProgram
import ru.yandex.market.mbi.orderservice.common.model.pg.PartnerEntity
import ru.yandex.market.mbi.orderservice.common.persistence.repository.pg.PartnerRepository
import ru.yandex.market.mbi.orderservice.tms.FunctionalTest
import java.time.LocalDateTime

class PartnerRepositoryTest : FunctionalTest() {

    @Autowired
    lateinit var partnerRepository: PartnerRepository

    @Test
    fun `test crud operations`() {
        val partner1 = PartnerEntity(
            partnerId = 100,
            businessId = 1000,
            name = "Test",
            organizationName = "Test",
            program = PlacementProgram.FBS
        )

        partnerRepository.upsert(partner1)
        val retrievedPartner1 = partnerRepository.findByPartnerId(100)
        assertThat(retrievedPartner1!!).satisfies {
            assertThat(it.partnerId).isEqualTo(100)
            assertThat(it.businessId).isEqualTo(1000)
            assertThat(it.name).isEqualTo("Test")
            assertThat(it.organizationName).isEqualTo("Test")
            assertThat(it.program).isEqualTo(PlacementProgram.FBS)
        }
        partnerRepository.upsert(partner1.copy(businessId = 1500, name = "Test2", updatedAt = LocalDateTime.now()))
        val retrievedPartner2 = partnerRepository.findByBusinessId(1500)[0]
        assertThat(retrievedPartner2).satisfies {
            assertThat(it.partnerId).isEqualTo(100)
            assertThat(it.businessId).isEqualTo(1500)
            assertThat(it.name).isEqualTo("Test2")
            assertThat(it.organizationName).isEqualTo("Test")
            assertThat(it.program).isEqualTo(PlacementProgram.FBS)
        }

        partnerRepository.upsert(
            partner1.copy(
                businessId = 1300,
                name = "Test3",
                updatedAt = LocalDateTime.now().minusDays(10)
            )
        )
        assertThat(partnerRepository.findByBusinessId(1300)).isEmpty()
        val retrievedPartner3 = partnerRepository.findByBusinessId(1500)[0]
        assertThat(retrievedPartner3).satisfies {
            assertThat(it.partnerId).isEqualTo(100)
            assertThat(it.businessId).isEqualTo(1500)
            assertThat(it.name).isEqualTo("Test2")
            assertThat(it.organizationName).isEqualTo("Test")
            assertThat(it.program).isEqualTo(PlacementProgram.FBS)
        }
    }
}
